scripts/2.TCGA-MLexample-covariates.py

```diff
@@ -281,66 +300,52 @@ def get_threshold_metrics(y_true, y_pred):
     for model, pipeline in final_pipelines.items()
 }
 
+coef_df = pd.concat([
+    get_model_coefficients(classifier, model, covariates.columns)
+    for model, classifier in final_classifiers.items()
+])
 
-# In[22]:
 
-def get_coefficients(classifier, X_mat):
-    coef_df = pd.DataFrame.from_items([
-        ('feature', X_mat.columns),
-        ('weight', classifier.coef_[0]),
-    ])
+# In[21]:
 
-    coef_df['abs'] = coef_df['weight'].abs()
-    coef_df = coef_df.sort_values('abs', ascending=False)
-    
-    return coef_df
+def zero(w): return '{:.1%}'.format(np.mean(w == 0))
+def negative(w): return '{:,}'.format(np.sum(w < 0))
+def positive(w): return '{:,}'.format(np.sum(w > 0))
 
-coef_df_dict = {
-    model: get_coefficients(classifier, X_partitions[model]['train'])
-    for model, classifier in final_classifiers.items()
-}
+coef_df.groupby('feature_set')['weight'].agg([zero, negative, positive])
 
 
-# In[23]:
+# In[22]:
 
 model = 'full'
-
-print('{:.1%} zero coefficients; {:,} negative and {:,} positive coefficients'.format(
-    (coef_df_dict[model].weight == 0).mean(),
-    (coef_df_dict[model].weight < 0).sum(),
-    (coef_df_dict[model].weight > 0).sum()
-))
-coef_df_dict[model].head(10)
+model_coef_df = coef_df[coef_df['feature_set'] == model]
+model_coef_df.head(10)
 
 
 # ## Investigate the predictions
 
-# In[24]:
+# In[23]:
 
 model = 'full'
 
-X_all = X_partitions[model]['train'].append(X_partitions[model]['test'])
-X_test_index = X_partitions[model]['test'].index
-y_all = y_train.append(y_test)
-
 predict_df = pd.DataFrame.from_items([
-    ('sample_id', X_all.index),
-    ('testing', X_all.index.isin(X_test_index).astype(int)),
-    ('status', y_all),
-    ('decision_function', final_pipelines[model].decision_function(X_all)),
-    ('probability', final_pipelines[model].predict_proba(X_all)[:, 1])
+    ('sample_id', X.index),
+    ('testing', X.index.isin(X_test.index).astype(int)),
+    ('status', y),
```

Comment:
May want to change the "testing" header to something like "in_testing_set". When I first saw this I thought the testing column was saying if the mutation was predicted as a 0 or 1 by the classifier.